scripts/doc-checker.py

```diff
@@ -0,0 +1,434 @@
+"""
+Validate docstrings using pydocstyle and numpydoc.
+
+Example usage:
+python scripts/doc-checker.py asv_bench/benchmarks/utils.py modin/pandas
+"""
+
+import argparse
+import pathlib
+import subprocess
+import os
+import ast
+from typing import List
+import sys
+import inspect
+from numpydoc.validate import Docstring
+
+MODIN_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.insert(0, MODIN_PATH)
+
+# error codes that pandas test in CI
+# https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks
+NUMPYDOC_BASE_ERROR_CODES = {
+    *("GL01", "GL02", "GL03", "GL05", "GL06", "GL07", "GL08", "GL09", "GL10"),
+    *("SS02", "SS03", "SS04", "SS05", "PR01", "PR02", "PR03", "PR04", "PR05"),
+    *("PR08", "PR09", "PR10", "RT01", "RT04", "RT05", "SA02", "SA03"),
+}
+
+MODIN_ERROR_CODES = {
+    "MD01": "'{parameter}' description should be '[type], default: [value]', found - '{found}'"
+}
+
+
+def get_optional_args(doc: Docstring) -> dict:
+    """
+    Get optional parameters for the object for which the docstring is checked.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Doctring
+        Docstring handler.
+
+    Returns
+    -------
+    dict
+        Dict with default arguments.
+    """
+    obj = doc.obj
+    if not callable(obj) or inspect.isclass(obj):
+        return None
+    signature = inspect.signature(obj)
+    return {
+        k: v.default
+        for k, v in signature.parameters.items()
+        if v.default is not inspect.Parameter.empty
+    }
+
+
+def validate_modin_error(doc: Docstring) -> list:
+    """
+    Validate custom modin errors.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    errors :
```

Comment:
numpydoc doesn't handle the case

Reply:
Maybe add a `RT02` check?

Reply:
Why do you use `errors :`? Return type should be `list`. The description explains what the list means.

Reply:
This is one of the available forms, but the user may forget to add a type, and numpydoc will not react to this.

```
-------
err_code : int
    Non-zero value indicates error code, or zero on success.
err_msg : str or None
    Human readable error message, or None on success.
```

Reply:
So far I just wrote a type without name.

Reply:
I don't mind on adding `RT02` if this makes sense.